README.md
# ELS - Enterprise Linux/EuroLinux Sources - libchamplain
 
## Usage:
  Checkout branch or tag. Use `make sources` to recreate  SOURCE0 archive.


src/ChangeLog
# Generated by Makefile. Do not edit.

commit 4ee5dca5d940bc9c75fe7e78c637325b9af28b50
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 16 19:49:11 2013 +0200

    0.12.4

commit 0631f5f06142839814ff886d9a9a515faed7d6c5
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 16 19:04:52 2013 +0200

    path-layer: Correct version since dashed lines are supported

commit ca218c19eb6b6edee49ad63071a8ef49d81b21e4
Merge: 8075e85 0ef0b54
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 16 18:24:27 2013 +0200

    Merge branch 'master' into libchamplain-0-12
    
    Conflicts:
    	configure.ac

commit 8075e85088f8c94763ead14920499f236635cd00
Merge: 8d8cdc9 0b79829
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 16 18:22:58 2013 +0200

    Merge commit '0b79829e2c56c37755006588cb557351315c63c7' into libchamplain-0-12
    
    Conflicts:
    	NEWS

commit 0ef0b5465be1ce8dc9721c1ad8d25e14df7e1fdc
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 16 17:51:34 2013 +0200

    factory: Define GTK_DISABLE_DEPRECATED for the deprecated sources

commit 50e2dcf0b0abd226b6fda901c3c4c93974283a9f
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 16 17:50:17 2013 +0200

    factory: Revert "Drop OAM map source completely"
    
    This reverts commit 5fbcda5c709d34efb87d6d60af1cf7f286e94fe1. After a second
    thought it's not a good idea to change the API by removing the define.
    Let's remove the define next time we make an API bump.
    
    Conflicts:
    	champlain/champlain-0.12.vapi

commit 5721b406df4bf176694f0d9d30793ee8c5853443
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 16 17:21:55 2013 +0200

    factory: Remove the Osmarender from the factory
    
    Osmarender doesn't exist any more and will be removed completely in the
    future version of libchamplain.

commit c69ae92180ebe65799a43ed65c2b3242311385ba
Author: Jiří Techet <techet@gmail.com>
Date:   Wed May 8 13:30:11 2013 +0200

    Convert some of the gints back to doubles because of lost precision

commit ce564e0cff5199c9b9ba66a2f05db94d5911c791
Author: Jiří Techet <techet@gmail.com>
Date:   Wed May 8 01:56:01 2013 +0200

    Fix the previous commit by adjusting offset based on the viewport size

commit a31db71dd4270bc68d80046d3538f7bf15319b1b
Author: Jiří Techet <techet@gmail.com>
Date:   Wed May 8 01:45:00 2013 +0200

    When zooming-in, recalculate viewport position from latitude/longitude to avoid rounding errors

commit 6a90f2766d21cbc17d69c1f9ee22f998db3f49e5
Author: Jiří Techet <techet@gmail.com>
Date:   Wed May 8 00:12:13 2013 +0200

    Compute zooming offset in double instead of int to avoid rounding errors

commit 9fa0a319aff8cf7bdd4b3bbf3cd25ad08034121e
Author: Jiří Techet <techet@gmail.com>
Date:   Tue May 7 23:53:37 2013 +0200

    Use sane color selection values when the obtained one aren't

commit 29002a75b648f3f5c7e09f69dc746b76fc057439
Author: Jiří Techet <techet@gmail.com>
Date:   Tue May 7 23:17:46 2013 +0200

    Remove background zoom actor when not needed any more

commit 99bf4aaa85bc0e873011b7f30ee9b1acebc74bc6
Author: Jiří Techet <techet@gmail.com>
Date:   Tue May 7 22:20:14 2013 +0200

    Check whether timeout exists before destroying it

commit 775c3f262805bc61a821dfcaa9e03245b3eb6609
Author: Jiří Techet <techet@gmail.com>
Date:   Tue May 7 21:56:41 2013 +0200

    Remove unused functions from ChamplainAdjustment and KineticScroll
    
    Also improve scrolling event handling slightly

commit 6dc2c4338f3e386472af104f833c3c3e19f0cd8f
Author: Jiří Techet <techet@gmail.com>
Date:   Tue May 7 21:51:22 2013 +0200

    Redraw using cairo on idle

commit 7325b3d91cc8104dc49adf5dd4689a57236a4ccf
Author: Jiří Techet <techet@gmail.com>
Date:   Sun May 5 19:31:46 2013 +0200

    Send the latitude/longitude signal less frequently and speed-up drawing

commit d8b95e483342625dc891bc334fa4debed805ce05
Author: Jiří Techet <techet@gmail.com>
Date:   Sat May 4 18:02:13 2013 +0200

    Check whether hadjustment & vadjustment exist before referencing them

commit 99beb25437346a81632cdb24d3ffaa8feedac5c5
Author: Jiří Techet <techet@gmail.com>
Date:   Sat May 4 17:08:05 2013 +0200

    Fix the view issues when it hasn't been realized yet

commit 382902bb40952508ff0e97f20628b9100c7467f4
Author: Jiří Techet <techet@gmail.com>
Date:   Sat May 4 14:09:38 2013 +0200

    Emit the relocation signal less frequently and speed-up viewport panning

commit 358a0256522106cc200b66134161d9ce071baa1b
Author: Jiří Techet <techet@gmail.com>
Date:   Fri May 3 19:50:28 2013 +0200

    Slightly improve zooming visualisation

commit b8a17b7b4069bfb2844a067b48c0d79d0018f136
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 2 22:29:59 2013 +0200

    Revert "Remove background zoom layer when all tiles loaded"
    
    This does not work as expected.

commit 0dc2fdfe28bf035bfabfd6b1f5047b92632e72e3
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 2 22:27:36 2013 +0200

    Update AUTHORS

commit 61f2e26977e4076d193958b6a3c5ccd80f3d1ee0
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 2 22:27:24 2013 +0200

    Update Clutter version in README

commit f5d26263bef757b20961d4dd318019eeb5f8ed39
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 2 22:27:08 2013 +0200

    Remove background zoom layer when all tiles loaded

commit 43978d828c530a11635c95a8989c6ead8384d5e3
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 2 21:27:28 2013 +0200

    Update the vapi file

commit 8b93ffbdb4b0115573cc711d417180ad95bd7760
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 2 21:16:08 2013 +0200

    Make sure ChamplainView is pixel-aligned
    
    Even when the view size is odd.

commit 8f73a29d16f2ddb54610ca3c37aca3099a81f737
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 2 18:20:55 2013 +0200

    Add more tile coordinate checks

commit 13fc7124b8da15343d4d327b03399c0284d94727
Author: Jiří Techet <techet@gmail.com>
Date:   Thu May 2 17:57:15 2013 +0200


src/champlain/champlain-version.h
/*
 * Copyright (C) 2009 Pierre-Luc Beaudoin <pierre-luc@pierlux.com>
 * File inspired by champlain-version.h.in which is
 * Authored By Matthew Allum  <mallum@openedhand.com>
 * Copyright (C) 2006 OpenedHand
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */

#ifndef __CHAMPLAIN_VERSION_H__
#define __CHAMPLAIN_VERSION_H__

/**
 * SECTION:champlain-version
 * @short_description: Versioning utility macros
 *
 * Champlain offers a set of macros for checking the version of the library
 * an application was linked to.
 */

/**
 * CHAMPLAIN_MAJOR_VERSION:
 *
 * The major version of libchamplain (1, if %CHAMPLAIN_VERSION is 1.2.3)
 */
#define CHAMPLAIN_MAJOR_VERSION   (0)

/**
 * CHAMPLAIN_MINOR_VERSION:
 *
 * The minor version of libchamplain (2, if %CHAMPLAIN_VERSION is 1.2.3)
 */
#define CHAMPLAIN_MINOR_VERSION   (12)

/**
 * CHAMPLAIN_MICRO_VERSION:
 *
 * The micro version of libchamplain (3, if %CHAMPLAIN_VERSION is 1.2.3)
 */
#define CHAMPLAIN_MICRO_VERSION   (4)

/**
 * CHAMPLAIN_VERSION:
 *
 * The full version of libchamplain, like 1.2.3
 */
#define CHAMPLAIN_VERSION         0.12.4

/**
 * CHAMPLAIN_VERSION_S:
 *
 * The full version of libchamplain, in string form (suited for
 * string concatenation)
 */
#define CHAMPLAIN_VERSION_S       "0.12.4"

/**
 * CHAMPLAIN_VERSION_HEX:
 *
 * Numerically encoded version of libchamplain, like 0x010203
 */
#define CHAMPLAIN_VERSION_HEX     ((CHAMPLAIN_MAJOR_VERSION << 24) | \
                                 (CHAMPLAIN_MINOR_VERSION << 16) | \
                                 (CHAMPLAIN_MICRO_VERSION << 8))

/**
 * CHAMPLAIN_CHECK_VERSION:
 * @major: major version, like 1 in 1.2.3
 * @minor: minor version, like 2 in 1.2.3
 * @micro: micro version, like 3 in 1.2.3
 *
 * Evaluates to %TRUE if the version of libchamplain is greater or equal
 * than @major, @minor and @micro
 */
#define CHAMPLAIN_CHECK_VERSION(major,minor,micro) \
        (CHAMPLAIN_MAJOR_VERSION > (major) || \
         (CHAMPLAIN_MAJOR_VERSION == (major) && CHAMPLAIN_MINOR_VERSION > (minor)) || \
         (CHAMPLAIN_MAJOR_VERSION == (major) && CHAMPLAIN_MINOR_VERSION == (minor) && CHAMPLAIN_MICRO_VERSION >= (micro)))

#endif /* __CHAMPLAIN_VERSION_H__ */


src/demos/markers-vala.c
/* markers-vala.c generated by valac 0.16.1, the Vala compiler
 * generated from markers-vala.vala, do not modify */

/*
 * Copyright (C) 2010 Simon Wenner <simon@wenner.ch>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */

#include <glib.h>
#include <glib-object.h>
#include <champlain/champlain.h>
#include <clutter/clutter.h>
#include <pango/pango.h>


#define TYPE_DEMO_LAYER (demo_layer_get_type ())
#define DEMO_LAYER(obj) (G_TYPE_CHECK_INSTANCE_CAST ((obj), TYPE_DEMO_LAYER, DemoLayer))
#define DEMO_LAYER_CLASS(klass) (G_TYPE_CHECK_CLASS_CAST ((klass), TYPE_DEMO_LAYER, DemoLayerClass))
#define IS_DEMO_LAYER(obj) (G_TYPE_CHECK_INSTANCE_TYPE ((obj), TYPE_DEMO_LAYER))
#define IS_DEMO_LAYER_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), TYPE_DEMO_LAYER))
#define DEMO_LAYER_GET_CLASS(obj) (G_TYPE_INSTANCE_GET_CLASS ((obj), TYPE_DEMO_LAYER, DemoLayerClass))

typedef struct _DemoLayer DemoLayer;
typedef struct _DemoLayerClass DemoLayerClass;
typedef struct _DemoLayerPrivate DemoLayerPrivate;
#define _g_object_unref0(var) ((var == NULL) ? NULL : (var = (g_object_unref (var), NULL)))
#define _g_error_free0(var) ((var == NULL) ? NULL : (var = (g_error_free (var), NULL)))

struct _DemoLayer {
	ChamplainMarkerLayer parent_instance;
	DemoLayerPrivate * priv;
};

struct _DemoLayerClass {
	ChamplainMarkerLayerClass parent_class;
};


static gpointer demo_layer_parent_class = NULL;

GType demo_layer_get_type (void) G_GNUC_CONST;
enum  {
	DEMO_LAYER_DUMMY_PROPERTY
};
DemoLayer* demo_layer_new (void);
DemoLayer* demo_layer_construct (GType object_type);


DemoLayer* demo_layer_construct (GType object_type) {
	DemoLayer * self = NULL;
	ClutterColor _tmp0_ = {0};
	ClutterColor orange;
	ChamplainLabel* _tmp1_;
	ChamplainLabel* _tmp2_;
	ChamplainLabel* marker;
	ChamplainLabel* _tmp3_;
	ChamplainLabel* _tmp4_;
	ChamplainLabel* _tmp5_;
	ClutterColor _tmp6_;
	ChamplainLabel* _tmp7_;
	ChamplainLabel* _tmp8_;
	ChamplainLabel* _tmp15_;
	ChamplainLabel* _tmp16_;
	ChamplainLabel* _tmp17_;
	ChamplainLabel* _tmp24_;
	ChamplainLabel* _tmp25_;
	ChamplainLabel* _tmp32_;
	ChamplainLabel* _tmp33_;
	ChamplainLabel* _tmp34_;
	GError * _inner_error_ = NULL;
	self = (DemoLayer*) g_object_new (object_type, NULL);
	_tmp0_.red = (guint8) 0xf3;
	_tmp0_.green = (guint8) 0x94;
	_tmp0_.blue = (guint8) 0x07;
	_tmp0_.alpha = (guint8) 0xbb;
	orange = _tmp0_;
	_tmp1_ = (ChamplainLabel*) champlain_label_new_with_text ("Montréal\n<span size=\"xx-small\">Québec</span>", "Serif 14", NULL, NULL);
	_tmp2_ = g_object_ref_sink (_tmp1_);
	marker = _tmp2_;
	_tmp3_ = marker;
	champlain_label_set_use_markup (_tmp3_, TRUE);
	_tmp4_ = marker;
	champlain_label_set_alignment (_tmp4_, PANGO_ALIGN_RIGHT);
	_tmp5_ = marker;
	_tmp6_ = orange;
	champlain_label_set_color (_tmp5_, &_tmp6_);
	_tmp7_ = marker;
	champlain_location_set_location ((ChamplainLocation*) _tmp7_, 45.528178, -73.563788);
	_tmp8_ = marker;
	champlain_marker_layer_add_marker ((ChamplainMarkerLayer*) self, (ChamplainMarker*) _tmp8_);
	{
		ChamplainLabel* _tmp9_;
		ChamplainLabel* _tmp10_;
		ChamplainLabel* _tmp11_;
		ChamplainLabel* _tmp12_;
		_tmp9_ = (ChamplainLabel*) champlain_label_new_from_file ("icons/emblem-generic.png", &_inner_error_);
		_tmp10_ = g_object_ref_sink (_tmp9_);
		_tmp11_ = _tmp10_;
		if (_inner_error_ != NULL) {
			goto __catch0_g_error;
		}
		_tmp12_ = g_object_ref_sink (_tmp11_);
		_g_object_unref0 (marker);
		marker = _tmp12_;
	}
	goto __finally0;
	__catch0_g_error:
	{
		GError* e = NULL;
		GError* _tmp13_;
		const gchar* _tmp14_;
		e = _inner_error_;
		_inner_error_ = NULL;
		_tmp13_ = e;
		_tmp14_ = _tmp13_->message;
		g_warning ("markers-vala.vala:40: %s", _tmp14_);
		_g_error_free0 (e);
	}
	__finally0:
	if (_inner_error_ != NULL) {
		_g_object_unref0 (marker);
		g_critical ("file %s: line %d: uncaught error: %s (%s, %d)", __FILE__, __LINE__, _inner_error_->message, g_quark_to_string (_inner_error_->domain), _inner_error_->code);
		g_clear_error (&_inner_error_);
		return NULL;
	}
	_tmp15_ = marker;
	champlain_label_set_text (_tmp15_, "New York");
	_tmp16_ = marker;
	champlain_location_set_location ((ChamplainLocation*) _tmp16_, 40.77, -73.98);
	_tmp17_ = marker;
	champlain_marker_layer_add_marker ((ChamplainMarkerLayer*) self, (ChamplainMarker*) _tmp17_);
	{
		ChamplainLabel* _tmp18_;
		ChamplainLabel* _tmp19_;
		ChamplainLabel* _tmp20_;
		ChamplainLabel* _tmp21_;
		_tmp18_ = (ChamplainLabel*) champlain_label_new_from_file ("icons/emblem-important.png", &_inner_error_);
		_tmp19_ = g_object_ref_sink (_tmp18_);
		_tmp20_ = _tmp19_;
		if (_inner_error_ != NULL) {
			goto __catch1_g_error;
		}
		_tmp21_ = g_object_ref_sink (_tmp20_);
		_g_object_unref0 (marker);
		marker = _tmp21_;
	}
	goto __finally1;
	__catch1_g_error:
	{
		GError* e = NULL;
		GError* _tmp22_;
		const gchar* _tmp23_;
		e = _inner_error_;
		_inner_error_ = NULL;
		_tmp22_ = e;
		_tmp23_ = _tmp22_->message;
		g_warning ("markers-vala.vala:50: %s", _tmp23_);
		_g_error_free0 (e);
	}
	__finally1:
	if (_inner_error_ != NULL) {
		_g_object_unref0 (marker);
		g_critical ("file %s: line %d: uncaught error: %s (%s, %d)", __FILE__, __LINE__, _inner_error_->message, g_quark_to_string (_inner_error_->domain), _inner_error_->code);
		g_clear_error (&_inner_error_);
		return NULL;
	}
	_tmp24_ = marker;
	champlain_location_set_location ((ChamplainLocation*) _tmp24_, 47.130885, -70.764141);
	_tmp25_ = marker;
	champlain_marker_layer_add_marker ((ChamplainMarkerLayer*) self, (ChamplainMarker*) _tmp25_);
	{
		ChamplainLabel* _tmp26_;
		ChamplainLabel* _tmp27_;
		ChamplainLabel* _tmp28_;
		ChamplainLabel* _tmp29_;
		_tmp26_ = (ChamplainLabel*) champlain_label_new_from_file ("icons/emblem-favorite.png", &_inner_error_);
		_tmp27_ = g_object_ref_sink (_tmp26_);
		_tmp28_ = _tmp27_;
		if (_inner_error_ != NULL) {
			goto __catch2_g_error;
		}
		_tmp29_ = g_object_ref_sink (_tmp28_);
		_g_object_unref0 (marker);
		marker = _tmp29_;
	}
	goto __finally2;
	__catch2_g_error:
	{
		GError* e = NULL;
		GError* _tmp30_;
		const gchar* _tmp31_;
		e = _inner_error_;
		_inner_error_ = NULL;
		_tmp30_ = e;
		_tmp31_ = _tmp30_->message;
		g_warning ("markers-vala.vala:59: %s", _tmp31_);
		_g_error_free0 (e);
	}
	__finally2:
	if (_inner_error_ != NULL) {
		_g_object_unref0 (marker);
		g_critical ("file %s: line %d: uncaught error: %s (%s, %d)", __FILE__, __LINE__, _inner_error_->message, g_quark_to_string (_inner_error_->domain), _inner_error_->code);
		g_clear_error (&_inner_error_);
		return NULL;
	}
	_tmp32_ = marker;
	champlain_label_set_draw_background (_tmp32_, FALSE);
	_tmp33_ = marker;
	champlain_location_set_location ((ChamplainLocation*) _tmp33_, 45.41484, -71.918907);
	_tmp34_ = marker;
	champlain_marker_layer_add_marker ((ChamplainMarkerLayer*) self, (ChamplainMarker*) _tmp34_);
	_g_object_unref0 (marker);
	return self;
}


DemoLayer* demo_layer_new (void) {
	return demo_layer_construct (TYPE_DEMO_LAYER);
}


static void demo_layer_class_init (DemoLayerClass * klass) {
	demo_layer_parent_class = g_type_class_peek_parent (klass);
}


static void demo_layer_instance_init (DemoLayer * self) {
}


GType demo_layer_get_type (void) {
	static volatile gsize demo_layer_type_id__volatile = 0;
	if (g_once_init_enter (&demo_layer_type_id__volatile)) {
		static const GTypeInfo g_define_type_info = { sizeof (DemoLayerClass), (GBaseInitFunc) NULL, (GBaseFinalizeFunc) NULL, (GClassInitFunc) demo_layer_class_init, (GClassFinalizeFunc) NULL, NULL, sizeof (DemoLayer), 0, (GInstanceInitFunc) demo_layer_instance_init, NULL };
		GType demo_layer_type_id;
		demo_layer_type_id = g_type_register_static (champlain_marker_layer_get_type (), "DemoLayer", &g_define_type_info, 0);
		g_once_init_leave (&demo_layer_type_id__volatile, demo_layer_type_id);
	}
	return demo_layer_type_id__volatile;
}



